src/main/kotlin/com/terraformation/backend/report/db/ReportStore.kt

```diff
@@ -298,32 +298,37 @@ class ReportStore(
   fun submit(reportId: ReportId) {
     requirePermissions { updateReport(reportId) }
 
-    ifLocked(reportId) {
-      val body =
-          dslContext
-              .select(REPORTS.BODY)
-              .from(REPORTS)
-              .where(REPORTS.ID.eq(reportId))
-              .fetchOne(REPORTS.BODY)!!
-              .let { objectMapper.readValue<ReportBodyModel>(it.data()) }
-              .toLatestVersion()
+    try {
+      ifLocked(reportId) {
+        val body =
+            dslContext
+                .select(REPORTS.BODY)
+                .from(REPORTS)
+                .where(REPORTS.ID.eq(reportId))
+                .fetchOne(REPORTS.BODY)!!
+                .let { objectMapper.readValue<ReportBodyModel>(it.data()) }
+                .toLatestVersion()
 
-      body.validate()
+        body.validate()
 
-      dslContext
-          .update(REPORTS)
-          .setNull(REPORTS.LOCKED_BY)
-          .setNull(REPORTS.LOCKED_TIME)
-          .set(REPORTS.STATUS_ID, ReportStatus.Submitted)
-          .set(REPORTS.SUBMITTED_BY, currentUser().userId)
-          .set(REPORTS.SUBMITTED_TIME, clock.instant())
-          .where(REPORTS.ID.eq(reportId))
-          .execute()
+        dslContext
+            .update(REPORTS)
+            .setNull(REPORTS.LOCKED_BY)
+            .setNull(REPORTS.LOCKED_TIME)
+            .set(REPORTS.STATUS_ID, ReportStatus.Submitted)
+            .set(REPORTS.SUBMITTED_BY, currentUser().userId)
+            .set(REPORTS.SUBMITTED_TIME, clock.instant())
+            .where(REPORTS.ID.eq(reportId))
+            .execute()
 
-      saveSeedBankInfo(body)
-      saveNurseryInfo(body)
+        saveSeedBankInfo(body)
+        saveNurseryInfo(body)
 
-      eventPublisher.publishEvent(ReportSubmittedEvent(reportId, body))
+        eventPublisher.publishEvent(ReportSubmittedEvent(reportId, body))
+      }
+    } catch (e: Exception) {
+      log.info("Report $reportId cannot be submitted: ${e.message}")
```

Comment:
Not part of this PR: Wondering if we should do an application-wide catch/log/throw.

Reply:
Exceptions that bubble up past the controller method are already logged (without stack traces, though, which maybe we should change). But this one was being caught by a try/catch in the controller method so the default logging didn't happen.

Auditing the controller methods and make sure any exceptions they catch get logged somewhere would probably be a good idea.